src/Adapter/Product/Repository/ProductMultiShopRepository.php

```diff
@@ -249,14 +249,17 @@ public function partialUpdate(Product $product, array $propertiesToUpdate, ShopC
     public function setCarrierReferences(ProductId $productId, array $carrierReferenceIds, ShopConstraint $shopConstraint): void
     {
         $shopIds = $this->getShopIdsByConstraint($productId, $shopConstraint);
+        $shopIdsIntegers = array_map(static function (ShopId $shopId): int {
+            return $shopId->getValue();
+        }, $shopIds);
```

Comment:
I thought about this instead. Ofc just a preference, so not blocking.
        $shopIds = array_map(static function (ShopId $shopId): int {
            return $shopId->getValue();
        }, $$this->getShopIdsByConstraint($productId, $shopConstraint));

Reply:
No, because `$shopIds` is used again a bit lower (although I admit the loop could be done with `$shopIdsIntegers`